src/rgw/rgw_rest_s3.h

```diff
@@ -873,6 +878,76 @@ static inline int valid_s3_bucket_name(const string& name, bool relaxed=false)
   return 0;
 }
 
+namespace s3selectEngine
+{
+class s3select;
+class csv_object;
+}
+
+class RGWSelectObj_ObjStore_S3 : public RGWGetObj_ObjStore_S3
+{
+
+private:
+  s3selectEngine::s3select *s3select_syntax;
+  std::string m_s3select_query;
+  std::string m_result;
+  s3selectEngine::csv_object *m_s3_csv_object;
+  std::string m_column_delimiter;
+  std::string m_quot;
+  std::string m_row_delimiter;
+  std::string m_compression_type;
+  std::string m_escape_char;
+  char * m_buff_header;
+  std::string m_header_info;
+  std::string m_sql_query;
+
+public:
+  unsigned int chunk_number;
+
+  enum header_name_En
+  {
+    EVENT_TYPE,
+    CONTENT_TYPE,
+    MESSAGE_TYPE
+  };
+  static const char *header_name_str[3];
+
+  enum header_value_En
+  {
+    RECORDS,
+    OCTET_STREAM,
+    EVENT
+  };
+  static const char *header_value_str[3];
+
+  RGWSelectObj_ObjStore_S3();
+  virtual ~RGWSelectObj_ObjStore_S3();
+
+  virtual int send_response_data(bufferlist &bl, off_t ofs, off_t len) override;
+
+  virtual int get_params() override;
+
+private:
+  void encode_short(char &buff, short s, int &i);
```

Comment:
char& buff, I guess.  Is it really used as a reference/address?  reference to char is very uncommon

Reply:
by mistake.